### PR TITLE
add pad_from_front param to pad_sequence

### LINKED ISSUE #10536: [feature request] Adding Pre and Post padding functionalities to pad_sequence function
I would like to add **pre** and **post** padding functionalities to ````torch.nn.utils.rnn.pad_sequence```` .
These functionalities haven't been included in the [function](https://pytorch.org/docs/stable/_modules/torch/nn/utils/rnn.html#pad_sequence)

It would be greatly beneficial for people using seq models like LSTM to have sequences created with both **pre** and **post** padding the sequences passed into the args. Other libraries like keras offer these functionalities in their APIs like: ````data = pad_sequences(data, maxlen=50, dtype='float32', padding='pre')````

Besides this, we can add another functionality to allow an argument(with optional value) to know the length of the sequences to be padded to. This is currently done with length of ````sequence[0]````(sequences passed in decreasing order of length).

Thanks!

cc @albanD @mruberry @jbschlosser

### PR DESCRIPTION
**Pull request by mskoh52**
Fixes #10536 

Adds a `pad_from_front: bool` argument to `torch.nn.utils.pad_sequence` which will stick the padding in front of the vector instead of the back.

### CODE REVIEW

**Comment by ezyang** [comment]
@jbschlosser you're our only hope

image

**Comment by jbschlosser** [comment]
Completely missed this one somehow, but willing to do a review if the author is still around and wants to start with a rebase!

**Comment by mskoh52** [comment]
Thanks for responding :) 

I'd forgotten about this and now I start to have some doubts about it. I realized later that I can just flip -> pad -> flip and have the same effect. Is there really a strong need to have this as a function argument?

Then there is the question of whether it should be as I've implemented here with `pad_from_front: bool` or `padding: Literal['pre', 'post']` as suggested in the linked issue.

But anyway, if folks think it's useful I'm happy to do a rebase and fix it up.

**Comment by jbschlosser** [comment]
> I'd forgotten about this and now I start to have some doubts about it. I realized later that I can just flip -> pad -> flip and have the same effect. In fact, that is what's done in the test. Is there really a strong need to have this as a function argument?

Fair enough - I don't see a strong reason to add this and there hasn't been a lot of activity asking for it recently :)

**Comment by dongin1009** [comment]
Previously, discussed from #10536 #61467 

If this issue still is considerable, I can make a pull request. I modified pad_sequence by adding a `pos` argument. This option will be necessary as same as the other ML frameworks support.

I just `flip` and compute the `dim`ensionality of tensor without manipulating the CPP file and tensor slicing.

```    
def pad_sequence(
    sequences: Union[Tensor, List[Tensor]],
    batch_first: bool = False,
    padding_value: float = 0.0,
    pos: str = 'right',
) -> Tensor:

    if pos='right':
        padded_sequence = torch._C._nn.pad_sequence(sequences, batch_first, padding_value)
    elif pos='left':
        sequences = tuple(map(lambda s: s.flip(0), sequences))
        padded_sequence = torch._C._nn.pad_sequence(sequences, batch_first, padding_value)
        _seq_dim = padded_sequence.dim()
        padded_sequence = padded_sequence.flip(-_seq_dim+batch_first)
    else:
        raise ValueError("pos should be either 'right' or 'left', but got {}".format(pos))
    return padded_sequence
```
